// @controlla/nuxt-typeform/dist/module.mjs
import { resolve } from 'path';
import { fileURLToPath } from 'url';
import { defineNuxtModule, addComponentsDir, addPlugin } from '@nuxt/kit';

const name = "@controlla/nuxt-typeform";
const version = "1.0.9";

const module = defineNuxtModule({
  meta: {
    name,
    version,
    configKey: "typeform",
    compatibility: {
      nuxt: "^3.0.0"
    }
  },
  defaults: {
    formId: ""
  },
  async setup(options, nuxt) {
    if (!options.formId) {
      await nuxt.close();
      throw new Error("You must provide a formId");
    }
    const runtimeDir = fileURLToPath(new URL("./runtime", import.meta.url));
    nuxt.options.build.transpile.push(runtimeDir);
    nuxt.options.css.push("@typeform/embed/build/css/popup.css");
    nuxt.options.runtimeConfig.public.typeform = {
      formId: options.formId
    };
    addComponentsDir({
      path: resolve(runtimeDir, "components"),
      pathPrefix: false,
      prefix: "",
      level: 999,
      global: true
    });
    addPlugin(resolve(runtimeDir, "plugin.client"));
    nuxt.options.build.transpile.push(runtimeDir, /@typeform\//);
  }
});

export { module as default };
